src/shared/utils/PermissionUtils.ts

```diff
@@ -35,5 +35,14 @@ export class PermissionUtils {
 
     await Database.put('Options', { key: 'notificationPermission', value: newPermission });
     OneSignalEvent.trigger(OneSignal.EVENTS.NOTIFICATION_PERMISSION_CHANGED_AS_STRING, newPermission);
+    
+    const newPermissionBoolean = newPermission === 'granted';
+    const previousPermissionBoolean = previousPermission === 'granted';
+
+    const shouldBeUpdatedBoolean = newPermissionBoolean !== previousPermissionBoolean || updateIfIdentical;
```

Comment:
can shorten to `shouldBeUpdated`. Also maybe add a comment to the function explaining what the argument `updateIfIdentical` does for quick reference.

Reply:
Fixed the naming on all these and split out the boolean even into it' own function to make things more clear.